google/resource_compute_network.go

```diff
@@ -64,6 +69,39 @@ func resourceComputeNetwork() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+
+			"peering": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateGCPName,
+						},
+						"network": &schema.Schema{
+							Type:             schema.TypeString,
+							Required:         true,
+							ValidateFunc:     validateRegexp(peerNetworkLinkRegex),
+							DiffSuppressFunc: peerNetworkLinkDiffSuppress,
```

Comment:
I could achieve the same thing by using a StateFunc that always saves the value to a canonical format. Any preferences?